aptos-move/framework/aptos-framework/sources/configs/Stake.move

```diff
@@ -28,41 +27,85 @@ module AptosFramework::Stake {
     const ESTAKE_TOO_HIGH: u64 = 6;
     /// Account is already a validator or pending validator.
     const EALREADY_VALIDATOR: u64 = 7;
-    /// Account is not a validator.
-    const ENOT_VALIDATOR: u64 = 8;
+    /// Account is not an active validator.
+    const ENOT_ACTIVE_VALIDATOR: u64 = 8;
     /// Can't remove last validator.
     const ELAST_VALIDATOR: u64 = 9;
     /// Delegation from the address already exists in this pool
     const EDELEGATION_ALREADY_EXIST: u64 = 10;
     /// Delegation to this stake pool will make it exceed tht maximum stake specified in ValidatorSet.
     const EDELEGATION_EXCEED_MAX: u64 = 11;
+    /// Account is not a validator.
+    const ENOT_VALIDATOR: u64 = 12;
+    /// Delegation found but for the wrong validator.
+    const EDELEGATION_EXISTS_WITH_DIFFERENT_VALIDATOR: u64 = 13;
+    /// Delegation is not active and cannot be unlocked.
+    const EDELEGATION_NOT_ACTIVE: u64 = 14;
+    /// Delegation is not yet marked as inactive and cannot be withdrawn.
+    const EDELEGATION_NOT_INACTIVE: u64 = 15;
+    /// Delegation amount is below the required minimum.
+    const EDELEGATION_AMOUNT_BELOW_MINIMUM: u64 = 16;
 
-    /// Basic unit of stake delegation, it's stored in StakePool.
-    struct Delegation has store {
-        coins: Coin<TestCoin>,
-        rewards: Coin<TestCoin>,
+    /// Delegation state enum values.
+    const DELEGATION_STATE_ACTIVE: u64 = 1;
+    const DELEGATION_STATE_INACTIVE: u64 = 2;
+    const DELEGATION_STATE_PENDING_ACTIVE: u64 = 3;
+    const DELEGATION_STATE_PENDING_INACTIVE: u64 = 4;
+
+    /// Represent a delegation to a staking pool. Shares represent % ownership of the pool, including rewards.
+    struct Delegation has key {
+        // validation_address and locked_until_secs should never change after a Delegation is created.
+        validator_address: address,
         locked_until_secs: u64,
+
+        // Shares and state can change after a Delegation is created.
+        shares: u64,
+        // Enum value.
+        state: u64,
+
+    }
+
+    /// Represent a pool of coins. Shares are issued according to the current share price (total coins / total shares).
+    struct CoinPool has store {
+        coins: Coin<TestCoin>,
+        total_shares: u64,
+    }
+
+    /// Wrapper around CoinPool to store the delegator addresses.
+    struct PendingCoinPool has store {
```

Comment:
why it's called Pending but stores addresses?

Reply:
Do you have a better name suggestion here? I was thinking it's pending delegations that have not been processed and we need addresses so that we can update the DelegationShare's state field when the pending delegations are processed at epoch boundary.